eox_core/api/v1/serializers.py

```diff
@@ -79,6 +89,86 @@ def validate(self, attrs):
         return attrs
 
 
+class EdxappExtendedUserSerializer(EdxappUserSerializer):
+    """
+    This serializer allows admiting all the
+    UserProfile fields and also the extra fields for the
+    userProfile.meta values that are specified in the
+    microsite settings.
+    """
+    year_of_birth = serializers.ChoiceField(choices=YEAR_OF_BIRTH_CHOICES)
+    gender = serializers.ChoiceField(choices=UserProfile.GENDER_CHOICES)
+    city = serializers.CharField()
+    goals = serializers.CharField()
+    bio = serializers.CharField(max_length=3000)
+    profile_image_uploaded_at = serializers.DateTimeField()
+    phone_number = serializers.CharField(max_length=50)
+    mailing_address = serializers.CharField()
+    courseware = serializers.CharField(max_length=255)
+    level_of_education = serializers.ChoiceField(
```

Comment:
are all these fields required?

Reply:
@mariajgrimaldi  Depends on the setting REGISTRATION_EXTRA_FIELDS. If you check line 129, when the serializer is initializing, the **required** attribute is modified according to what is defined in the settings

Reply:
I think maybe I should explain this in the serializer docstring